File: src/main/resources/public/app/models/credentials/user-credentials.js
/**
 * Value object, containing user credentials.
 */
export default class UserCredentials {

  /**
   * User login.
   * @type {Login}.
   * @readonly
   */
  login;

  /**
   * User password.
   * @type {Password}.
   * @readonly
   */
  password;

  /**
   * Instantiates UserCredentials value object.
   *
   * @param {Login} login - user login.
   * @param {Password} password - password.
   */
  constructor(login, password) {
    this.login = login;
    this.password = password;
    Object.freeze(this);
  }
}

File: src/main/java/ua/nure/sealthenote/models/userCredentails/UserCredentials.java
package ua.nure.sealthenote.models.userCredentails;

public class UserCredentials {

    private final String login;
    private final String password;

    public UserCredentials(String login, String password) {
        this.login = login;
        this.password = password;
    }

    public String login() {
        return login;
    }

    public String password() {
        return password;
    }
}
